### DISCUSSION TITLE
What's the recommended way to throw exceptions inside a native function?

### DISCUSSION THREAD

**Discussion by Swimburger** [microsoft/semantic-kernel#2760] [Q&A] [answered]
Should a native function return the error message as a string without an exception? Or throw normal .NET exceptions?

**Comment by Swimburger** [comment, marked as answer]
Based on the discussion of todays office hour: Throw native .NET exceptions as you normally would in .NET programs.

**Reply by jasonmalinowski** [reply]
I was about to ask this same question but found this discussion -- was there discussion of why to do this? Other than it can make the code cleaner, does Semantic Kernel do other things with the exception?

**Reply by ashahabov** [reply]
I also have the same question. My code validates passing parameters. Should I return this validation error list as a string collection or throw an exception? Or maybe it is better to create a custom exception with a property `AppMessage` and in the agent instructions ask to read this message to understand why the parameter value is invalid?
